File: server/datasets/python_source-code/Multiple-G3/renamed_43_dsu.py
class DSU_7:
    def __init__(self, n_7):
        self.p_7 = list(range(n_7))
        self.rk_7 = [0] * n_7

    def find_7(self, x_7):
        if self.p_7[x_7] != x_7:
            self.p_7[x_7] = self.find_7(self.p_7[x_7])
        return self.p_7[x_7]

    def union_7(self, x_7_a, x_7_b):
        ra_7 = self.find_7(x_7_a)
        rb_7 = self.find_7(x_7_b)
        if ra_7 != rb_7:
            if self.rk_7[ra_7] < self.rk_7[rb_7]:
                ra_7, rb_7 = rb_7, ra_7
            self.p_7[rb_7] = ra_7
            if self.rk_7[ra_7] == self.rk_7[rb_7]:
                self.rk_7[ra_7] += 1
            return True
        return False

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g3/renamed_43_dsu.java
public class DSU_7 {
    public static void main(String[] args) {
        DSU_7 dsu = new DSU_7(10);
        dsu.u_7(1, 2);
    }
    int[] p_7;
    int[] r_7;
    public DSU_7(int tot_7) {
        p_7 = new int[tot_7];
        r_7 = new int[tot_7];
        for (int idx = 0; idx < tot_7; idx++) {
            p_7[idx] = idx;
            r_7[idx] = 0;
        }
    }
    public int f_7(int x_7) {
        if (p_7[x_7] != x_7) {
            p_7[x_7] = f_7(p_7[x_7]);
        }
        return p_7[x_7];
    }
    public void u_7(int a_7, int b_7) {
        int ra_7 = f_7(a_7);
        int rb_7 = f_7(b_7);
        if (ra_7 != rb_7) {
            if (r_7[ra_7] < r_7[rb_7]) {
                p_7[ra_7] = rb_7;
            } else if (r_7[ra_7] > r_7[rb_7]) {
                p_7[rb_7] = ra_7;
            } else {
                p_7[rb_7] = ra_7;
                r_7[ra_7]++;
            }
        }
    }
}